packages/go_router/lib/src/router.dart

```diff
@@ -299,6 +302,15 @@ class GoRouter implements RouterConfig<RouteMatchList> {
   @override
   late final GoRouteInformationParser routeInformationParser;
 
+  /// When set to [true], the route returned by `getInitialRoute` will take precedence over the
+  /// platform's default initial location. This allows developers to control the starting route
+  /// of the application independently of the platform.
+  ///
```

Comment:
somewhere in the doc should talk about when platform's default initial location will be set. e.g. deeplink

Reply:
added these docs